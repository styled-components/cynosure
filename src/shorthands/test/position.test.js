// @flow
import position from '../position'

describe('position', () => {
  it('properly applies a value when passed only one', () => {
    expect(position('relative', '12px')).toEqual({
      bottom: '12px',
      left: '12px',
      position: 'relative',
      right: '12px',
      top: '12px',
    })
  })
  it('properly applies values when passed two', () => {
    expect(position('relative', '12px', '24px')).toEqual({
      bottom: '12px',
      left: '24px',
      position: 'relative',
      right: '24px',
      top: '12px',
    })
  })
  it('properly applies values when passed three', () => {
    expect(position('relative', '12px', '24px', '36px')).toEqual({
      bottom: '36px',
      left: '24px',
      position: 'relative',
      right: '24px',
      top: '12px',
    })
  })
  it('properly applies values when passed four', () => {
    expect(position('relative', '12px', '24px', '36px', '48px')).toEqual({
      bottom: '36px',
      left: '48px',
      position: 'relative',
      right: '24px',
      top: '12px',
    })
  })
  it('properly ignores position property, when not passed one', () => {
    expect(position('12px', '24px', '36px', '48px')).toEqual({
      bottom: '36px',
      left: '48px',
      right: '24px',
      top: '12px',
    })
  })
  it('properly skips values when passed undefined`', () => {
    expect(position('relative', '12px', null, '36px', '48px')).toEqual({
      bottom: '36px',
      left: '48px',
      position: 'relative',
      top: '12px',
    })
  })
  it('properly skips first value when passed undefined', () => {
    expect(position(null, '24px', '36px', '48px')).toEqual({
      bottom: '36px',
      left: '48px',
      right: '24px',
    })
  })
  it('properly applies values when passed four integers', () => {
    expect(position('relative', 12, 24, 36, 48)).toEqual({
      bottom: 36,
      left: 48,
      position: 'relative',
      right: 24,
      top: 12,
    })
  })
  it('properly applies zero value', () => {
    expect(position('relative', 0)).toEqual({
      bottom: 0,
      left: 0,
      position: 'relative',
      right: 0,
      top: 0,
    })
  })
})
