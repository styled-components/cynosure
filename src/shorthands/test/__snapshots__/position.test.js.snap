exports[`position properly applies a value when passed only one 1`] = `
Object {
  "bottom": "12px",
  "left": "12px",
  "position": "relative",
  "right": "12px",
  "top": "12px",
}
`;

exports[`position properly applies unitless values 1`] = `
Object {
  "bottom": 36,
  "left": 48,
  "right": 24,
  "top": 12,
}
`;

exports[`position properly applies unitless zero 1`] = `
Object {
  "bottom": 0,
  "left": 0,
  "right": 0,
  "top": 0,
}
`;

exports[`position properly applies values when passed four 1`] = `
Object {
  "bottom": "36px",
  "left": "48px",
  "position": "relative",
  "right": "24px",
  "top": "12px",
}
`;

exports[`position properly applies values when passed three 1`] = `
Object {
  "bottom": "36px",
  "left": "24px",
  "position": "relative",
  "right": "24px",
  "top": "12px",
}
`;

exports[`position properly applies values when passed two 1`] = `
Object {
  "bottom": "12px",
  "left": "24px",
  "position": "relative",
  "right": "24px",
  "top": "12px",
}
`;

exports[`position properly ignores position property, when not passed one 1`] = `
Object {
  "bottom": "36px",
  "left": "48px",
  "right": "24px",
  "top": "12px",
}
`;
